CCD of NI Unit files - Gain Control, Trigger Generator, daqmx files

krishmaj opened the pull request:
@jackdunger Can you take a look at this PR to make sure I didn't accidentally muck anything up?  I'm not going to merge it myself, unless you give me the all-clear to.

- Renamed "ni_box" script to "ni_trigger_generator", and added documentation and config quantities.
- Renamed "gain_control" script to "ni_gain_control", redid the code flow according to our earlier discussion, and added extensive documentation (since I think this one is possibly the most potentially confusing!) and config quantities.
- Cleaned up and corrected some typos in the /daqmx folder files, but (hopefully!) have not accidentally changed the functionality.

The only remaining things I need to look at are the /server and /interlock files, and adding function documentation to the smellie_controller one.  I will do those over the weekend.

jackdunger commented on smellie/daqmx/config.py:
```diff
@@ -2,24 +2,20 @@
-    # Full path of the NIDAQmx.h file
-    # Default location on Windows XP
+    # This is the full path of the NIDAQmx.h file if it is installed on a Windows XP machine, which (as of 29/04/16) is also the default location
     dot_h_file =  r'C:\Program Files (x86)\National Instruments\NI-DAQ\DAQmx ANSI C Dev\include\NIDAQmx.h'
```
let's move these to the config file?

krishmaj replied:
Will do - just to be clear, this config file would still remain, and would contain only the checks for the platform and/or architecture?

krishmaj commented:
@jackdunger I've added a commit to this PR for the Interlock, Server and Controller files.  I had expected to do these in a separate PR, but I mucked up my Git branching - sorry!
- interlock/keep_alive.py : purely documentation and cleanup
- server/exception_handler.py : documentation and cleanup, and I've changed what I think is some incorrect logic in the "if" statement around line 20 ... if I'm reading it correctly, the unchanged code would have always returned an unhandled exception, even if it was handled (in which case it would have also returned as handled as well).
- server/smellie_server.py : documentation and cleanup, and I removed a (I think!) unnecessary double-bracket on old line 17.
- added documentation for all functions, renamed a few existing ones to be more clear and added a template one for the SuperK_Master_Mode.  I've also added the Fibre Switch into the deactivate function - we are not using outputs 13 and 14 on the FS (no fibres connected), so one of them can and should be used in deactivated mode.

jackdunger commented:
Hi @krishmaj sorry I forgot about this! Thank you for the changes, I fixed the conflicts and merged this in
